recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
@martinb3 it seems to me that a library fits more when trying to configure a single, simple entity (a file, a user but not a group of things along with many aspects like sudo,shadow in this case ) but apart from that (which is a personal preference) it feels that it will require that the consumer will have to dig at the code of the library anyway to understand how to construct a block that gives them the desired result. In that respect I would prefer if we added the whole thing into a template and exposed it completely to the consumer, keeping the 'any' logic as default and using the data to drive the membership. I like the cookbook because it also documents things better (e.g. README examples), we can push improvements to consumers and also we benefit from feedback from the community.